### DISCUSSION TITLE
reduction and its dimensions to use  and optimal number of cluster

### DISCUSSION THREAD

**Discussion by pengxin2019** [stuart-lab/signac#808] [Q&A] [answered]
Hi Tim,
Thanks for developing this great tool.
I am using your pipeline to integrate 4 scATAC datasets following your desmonstrated pipeline here https://satijalab.org/signac/articles/integrate_atac.html

here is the merged object and my code to integrate:
```
# > pbmc.combined
# An object of class Seurat 
# 101942 features across 17235 samples within 1 assay 
# Active assay: peaks (101942 features, 101942 variable features)
# 1 dimensional reduction calculated: lsi

integration.anchors <- FindIntegrationAnchors(
  object.list = list(pbmc.atac.6798.1x, pbmc.atac.6800.1x,pbmc.atac.6798.2x,pbmc.atac.6800.2x),
  anchor.features = granges(common.peaks),
  reduction = "rlsi",
  dims = 2:30
)
integrated <- IntegrateEmbeddings(
  anchorset = integration.anchors,
  reductions = pbmc.combined[["lsi"]],
  new.reduction.name = "integrated_lsi",
  dims.to.integrate = 1:30
)

integrated <- RunUMAP(integrated, reduction = "integrated_lsi", dims = 2:30)

integrated <- RunTFIDF(integrated, min.cutoff = 'q0') %>% FindTopFeatures(min.cutoff = 10) %>% RunSVD()

```
Then this integrated look like this:
```
# > integrated
# An object of class Seurat 
# 101942 features across 17235 samples within 1 assay 
# Active assay: peaks (101942 features, 101942 variable features)
# 3 dimensional reductions calculated: integrated_lsi, umap, lsi
```
I have a couple of quesitions:
1. in terms of the `dims.to.integrate` of `IntegrateEmbeddings` is set up as 1:30 rather than 2:30? whose dimension is being talked about here?  1:30 of `pbmc.combined[["lsi"]]`?  I think you mentioned usually 1st dimension is highly correlated with sequencing depth, right?
2. as you can see that this `integrated` object has an `integrated_lsi` reductions since I have ran `IntegrateEmbeddings`  and it also has an `lsi` reduction since I have ran `RunTFIDF`. But which one is the most accurate one  and should be used in the downstream analysis? like FindNeighbors and FindClusters? 

3. I noticed that only the standard deviation of `lsi`   is calculated. Do you think the elbow  plot below plotted using `lsi` is transferrable to  `integrated_lsi` reduction if  the answer for Q2 is `integrated_lsi` reduction is more accurate? 
As you can see from below,  sd is pretty small when around dims 15. Do you think that we should use 2:15 in UMAP instead of 2:30? Or acutually that using more dim would never hurt? 2:30 is more accurate than 2:15 and 2:50 is more accurate than 2:30?

By the way, do you think the calculation of Eclidean distance between clusters proposed in this post https://github.com/satijalab/seurat/issues/1419 is helpful in terms of optimal number of clusters to create? since the sum of distance between clusters with a optimal number should be lowest than that of other number of clusters?

Thanks
Best
Penny
[p.merged.ElbowPlot.pdf](https://github.com/timoast/signac/files/7264637/p.merged.ElbowPlot.pdf)

**Comment by timoast** [comment, marked as answer]
> in terms of the dims.to.integrate of IntegrateEmbeddings is set up as 1:30 rather than 2:30? whose dimension is being talked about here? 1:30 of pbmc.combined[["lsi"]]?

Yes, this is the dimension reduction specified by the `reductions` parameter. You can use 2:30 if you prefer, but the integrated reduction that is returned will only have those dimensions present (ie, use integrated dimensions 1:29 for original dimensions 2:30). See the documentation for IntegrateEmbeddings: https://satijalab.org/seurat/reference/integrateembeddings

> as you can see that this integrated object has an integrated_lsi reductions since I have ran IntegrateEmbeddings and it also has an lsi reduction since I have ran RunTFIDF. But which one is the most accurate one and should be used in the downstream analysis? like FindNeighbors and FindClusters?

One is integrated (differences between batches removed), the other is the original dimension reduction.

>  Do you think the elbow plot below plotted using lsi is transferrable to integrated_lsi reduction

Yes

> Do you think that we should use 2:15 in UMAP instead of 2:30? Or acutually that using more dim would never hurt? 2:30 is more accurate than 2:15 and 2:50 is more accurate than 2:30? 

See https://satijalab.org/signac/articles/faq.html#how-should-i-decide-on-the-number-of-dimensions-to-use

> By the way, do you think the calculation of Eclidean distance between clusters proposed in this post satijalab/seurat#1419 is helpful in terms of optimal number of clusters to create? 

No I don't think so. You could try looking at the Silhouette metric: https://en.wikipedia.org/wiki/Silhouette_(clustering)

**Reply by pengxin2019** [reply]
1.
_Penny--Do you think the elbow plot below plotted using lsi is transferrable to integrated_lsi reduction

Tim--Yes_

Since there is an elbow plot for the `lsi` reduction and no elbow plot for `integrated_lsi` reduction.
Lets say for example, the standard deviation explained by the `2:30 `dimentions of `lsi` is 60%. do you think it is fair to assume that the standard deviation explained by `2:30` dimentions of`integrated_lsi`  reduction is also 60% or close to 60%? 

Also, do you think it is possible that if  the number of dimensions included in the `UMAP` is too high, some noise might be included in the `UMAP` to confuse us? Why did you choose to use` 2:30` rather than `2:50` to make the `UMAP`?

2. 
```
integrated <- IntegrateEmbeddings(
  anchorset = integration.anchors,
  reductions = pbmc.combined[["lsi"]],
  new.reduction.name = "integrated_lsi",
  dims.to.integrate = 1:30
)
```
In the code above, you used `1:30 ` rather than `2:30` is the number of dimension of `pbmc.combined` object to identify the integrateembeddings, does that mean `pbmc.combined` is not as highly correlated to sequencing depth as that of `integrated` object? I remember in your demonstrated pipeline of scATAC ananlysis, you chose to use 2:30 since 1st dimension is highly correlated with sequencing depth.

3. It seems like in the documentation of functions of Seurat and Signac, the `slot`  used by each function is not mentioned. Do most of the functions use `data` slot rather than the `counts` which is the raw data from 10x or `scale.data`? I noticed only a few functions require `scale.data` slot
Thanks
Best
penny

**Reply by timoast** [reply]
> Lets say for example, the standard deviation explained by the 2:30 dimentions of lsi is 60%. do you think it is fair to assume that the standard deviation explained by 2:30 dimentions ofintegrated_lsi reduction is also 60% or close to 60%?

The dimension reduction is the same, there's just been an adjustment made in the integrated version to correct for differences between the batches. You should use the same dimensions for the integrated and original versions of the dimension reduction.

> It seems like in the documentation of functions of Seurat and Signac, the slot used by each function is not mentioned. Do most of the functions use data slot rather than the counts which is the raw data from 10x or scale.data? I noticed only a few functions require scale.data slot

The slot used will depend on the function. Raw counts is used as input for data normalization. Normalized data is input for data scaling. Scaled data is used for PCA. Normalized data is used for most differential expression tests, although not all. Most visualizations used normalized data, except heatmaps use scaled data. For scATAC, scaled data is not used.
